### ISSUE TITLE
Responsive element does not display on Motorola MC40

### ISSUE THREAD

**Issue by Matt-Butler** [cerner/terra-core#304]
Datepicker does not work on Motorola MC40

Mobile browser 30.30.30
Kitkat - Android 4.4.3

**Comment by Matt-Butler** [comment]
https://developers.google.com/web/tools/chrome-devtools/remote-debugging/

This document has information on remote debugging android devices

**Comment by Matt-Butler** [comment]
MC40 console output (production env)

**http://engineering.cerner.com/terra-core/#/site**
- no console output 

**http://engineering.cerner.com/terra-core/#/site/demographics-banner**
<img width="1347" alt="screen shot 2017-05-10 at 10 12 06 am" src="https://cloud.githubusercontent.com/assets/8146159/25906067/4fe0f3b8-3569-11e7-97a0-90d51645c821.png">

**http://engineering.cerner.com/terra-core/#/site/datepicker**
<img width="1377" alt="screen shot 2017-05-10 at 10 07 12 am" src="https://cloud.githubusercontent.com/assets/8146159/25905772/7c663430-3568-11e7-811d-b15f78819321.png">

I'll try again in a dev environment so I can get more information on the console errors

**Comment by Matt-Butler** [comment]
MC40 console output (development env)

**http://engineering.cerner.com/terra-core/#/site**
- no console output 

**http://{myIpAddress}/#/site/demographics-banner**
<img width="1100" alt="screen shot 2017-05-10 at 10 22 14 am" src="https://cloud.githubusercontent.com/assets/8146159/25906581/afe80e1c-356a-11e7-943a-a0a6e3176091.png">


**http://{myIpAddress}/#/site/datepicker**
<img width="1106" alt="screen shot 2017-05-10 at 10 24 02 am" src="https://cloud.githubusercontent.com/assets/8146159/25906633/ce07473c-356a-11e7-9082-39345af24a12.png">

**Comment by bjankord** [comment]
Hmm, does the datepicker use the `<ResponsiveElement />`? Seeing the getBreakpoints and handleResize methods makes me think its something related to that component. Might test this page and see if it renders on the MC40: http://engineering.cerner.com/terra-core/#/site/responsive-element

**Comment by Matt-Butler** [comment]
I opened each of the component pages. I did not interact with any of the components. 
- [x] = no console errors

---

- [x] Arrange
- [x] Base
- [x] Badge
- [x] Button
- [x] Button Group
- [x] Content
- [x] Content Container
- [ ] Datepicker
- [ ] Demographics Banner
- [x] Grid
- [x] I18n
- [x] Icon
- [x] Image
- [x] List
- [x] Modal
- [x] Progress Bar
- [ ] Responsive Element
- [x] Slide Panel
- [x] Standout
- [x] Status
- [x] Table
- [x] Title

**Comment by Matt-Butler** [comment]
**http://{myIpAddress}/#/site/responsive-element**
<img width="1102" alt="screen shot 2017-05-10 at 10 34 42 am" src="https://cloud.githubusercontent.com/assets/8146159/25907313/b3af62d2-356c-11e7-98e6-7dada4cb588f.png">

**Comment by Matt-Butler** [comment]
> Hmm, does the datepicker use the <ResponsiveElement />? Seeing the getBreakpoints and handleResize methods makes me think its something related to that component. Might test this page and see if it renders on the MC40: http://engineering.cerner.com/terra-core/#/site/responsive-element


Yes, It is the responsive element component that does not work on the motorola MC40

**Comment by Matt-Butler** [comment]
Both of the other packages have the responsive element as a dependency:

- https://github.com/cerner/terra-core/blob/master/packages/terra-demographics-banner/package.json#L35

- https://github.com/cerner/terra-core/blob/master/packages/terra-date-picker/package.json#L34

**Comment by Matt-Butler** [comment]
The resize-observer-polyfill only supports **android 5.1 and above.**

![screen shot 2017-05-10 at 10 42 11 am](https://cloud.githubusercontent.com/assets/8146159/25907557/6015d466-356d-11e7-91f6-210fa96c2b0b.png)
Taken from: https://github.com/que-etc/resize-observer-polyfill#browser-support

**Comment by Matt-Butler** [comment]
This is the fuction it is failing on:
```
/**
 * Translates SASS defined breakpoints into a javascript accessible object
 * @returns {Object} - Returns SASS defined breakpoints in pixels units as key value pairs
 */
var getBreakpoints = function getBreakpoints() {
  if (breakpoints) {
    return breakpoints;
  }

  var data = void 0;
  var datasource = document.createElement('div');
  datasource.className = 'terra-Breakpoints';

  document.body.appendChild(datasource);

  data = window.getComputedStyle(datasource, ':before').getPropertyValue('content');
  data = data.replace(/^['"]+|\s+|\\|(;\s?})+|['"]$/g, '');

  breakpoints = JSON.parse(data);

  document.body.removeChild(datasource);

  return breakpoints;
};
```
The error is thrown on this line because `data` is an empty string.
```
 breakpoints = JSON.parse(data);
```

**Comment by Matt-Butler** [comment]
If you hard code the breakpoints as
```
breakpoints = {"tiny":544,"small":768,"medium":992,"large":1216,"huge":1440};
```

It fixes the issue

**Comment by Matt-Butler** [comment]
We could do something like this temporarily until we find a better way to convert sass -> json
```
  try {
      breakpoints = JSON.parse(data);
  } catch(e) {
      console.error(e);
      console.error(`JSON.parse(data) failed. data = ${data}`);
      console.info(`If you have custom breakpoints, the values may be incorrect on this device`);
      breakpoints = {"tiny":544,"small":768,"medium":992,"large":1216,"huge":1440};
  }
```

**Comment by Matt-Butler** [comment]
```
    data = window.getComputedStyle(datasource, ':before').getPropertyValue('content');
    window.data = data;
```


MC40:
```
console.log(data);

// returns undefined
```

Chrome (current):
```
console.log(data);

// returns  ""{\"tiny\":544,\"small\":768,\"medium\":992,\"large\":1216,\"huge\":1440}""
```

**Comment by bjankord** [comment]
Resolved with #333

**Comment by bjankord** [comment]
> The resize-observer-polyfill only supports android 5.1 and above. 

Interesting, @dkasper-was-taken was mentioning using this I believe with the menu component. We may want to look into this more if we plan to use the resize-observer for other components.
